### ISSUE TITLE
Transformer Big and layer_prepostprocess_dropout

### ISSUE THREAD

**Issue by ColainCYY** [OpenNMT/OpenNMT-py#885]
I have got a pretty BLEU-result by transformer-NMT on my own data with the below commands (core params):

```
-src_word_vec_size 512 -tgt_word_vec_size 512 -rnn_size 512 -position_encoding \
-encoder_type transformer -decoder_type transformer -enc_layers 6 -dec_layers 6 \
-self_attn_type scaled-dot -label_smoothing 0.1 -dropout 0.1 -param_init_glorot \
-adam_beta2 0.997 -decay_method noam -batch_size 40 -max_grad_norm 0 -param_init 0 \
-warmup_steps 8000 -seed 1 -optim adam -learning_rate 2
```

as described in the paper, size of 512 is the base-transformer, and the big-transformer performes better. my big-transformer is run as:
```
-src_word_vec_size 1024 -tgt_word_vec_size 1024 -rnn_size 1024 -heads 16 \
-transformer_ff 4096 -position_encoding -encoder_type transformer -decoder_type transformer \
-enc_layers 6 -dec_layers 6 -self_attn_type scaled-dot -label_smoothing 0.1 -dropout 0.3 \
-warmup_steps 16000 -accum_count 1 -seed 1 -optim adam -learning_rate 2 -adam_beta2 0.997 \
-decay_method noam -param_init_glorot
```

to my disappointment, my BLEU is even lower than the base model. I look up to the google-tensor2tensor code for answer, and found that there is an extra param "layer_prepostprocess_dropout", which is different between base-model and big-model. 

tensor2tensor insert a dropout layer between embedding layer and stacked attention layers. In OpenNMT, I didn't found the same settings.

My two questions are,

why the big-model doesn't outperform the base-model?
why the prepostprocess-dropout layer is missing?

Thanks for any help. It has confused me for a long time.   ...>___<...

**Comment by vince62s** [comment]
I think you may need to adjust a bit.
Even for the base, using token batches seem to give better results.
instead of batch_size 40 (sentences) try in token mode with 4096.

Now, for the big, you won't be able to use 4096 tokens, you may have to reduce a bit.
did you omit purposely max_grad_norm in the big config ? also param_init 0 ?

for both,if you use only 1 GPU, I suggest you use accum_count 4

**Comment by ColainCYY** [comment]
Thanks for your suggestion~ @vince62s 
I train my model with 8 GPUs. now, both the two model are sentence-batchde. 
Within the same settings, but I have gotten a worse big model than the base model. So confused.

**Comment by vince62s** [comment]
did you omit purposely max_grad_norm in the big config ? also param_init 0 ?
or is the command line you displayed for big is incomplete ?

**Comment by ColainCYY** [comment]
@vince62s sorry, I didn't paste the complete line. I use "-param_init 0 -max_grad_norm 0" as the base model. I wonder whether my data is large enough for the big-model to show its difference from the base model. my data consists of 2 million para-lines.

**Comment by ColainCYY** [comment]
I have also tried add a layer of prepostprocess_dropout as tensor2tensor code. But it didn't work.
